moshi/src/main/java/com/squareup/moshi/Moshi.java

```diff
@@ -35,14 +36,14 @@
   private final Map<Object, JsonAdapter<?>> adapterCache = new LinkedHashMap<>();
 
   private Moshi(Builder builder) {
-    List<JsonAdapter.Factory> factories = new ArrayList<>();
+    Set<JsonAdapter.Factory> factories = new LinkedHashSet<>();
```

Comment:
This won't work, because the order will be wrong for factories like `StandardJsonAdapters.FACTORY`.